src/Framework/PaymentGateways/PaymentGateway.php

```diff
@@ -95,8 +95,8 @@ public function handleCreatePayment(Donation $donation)
              * @unreleased
              */
             $gatewayData = apply_filters(
-                "givewp_new_payment_{$donation->gatewayId}_gateway_data",
-                null,
+                "givewp_create_payment_gateway_data_{$donation->gatewayId}",
+                !empty($_REQUEST['gatewayData']) ? give_clean($_REQUEST['gatewayData']) : [],
```

Comment:
@ravinderk @JasonTheAdams something I just noticed.. 

In Next Gen, we are posting data to the server as `application/json` not formData like wp-ajax does, so we need to use `file_get_contents('php://input')` to parse the request data.

Checking for this superglobal key won't work in that situation.

Reply:
commit: https://github.com/impress-org/givewp/compare/f5c2d6d6e806...a24bff041e10